### PR TITLE
chore(tsconfig): add base tsconfig.json

### PR DESCRIPTION
**Pull request by mayorovad**
**Summary**
Add base `tsconfig.json` to root of repository and extend packages upon it. Refactoring `src` values of `tsconfig.json` of packages.

**Description for the changelog**
- Moved repeating declarations in base `tsconfig.json`
- Removed `ESNext` target/module/lib from `ts-example`, which will lead to use of `es2020` from base `tsconfig.json`

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "compilerOptions": {
+    "module": "es2020",
+    "lib": ["dom", "es2020"],
+    "target": "es2020",
```
Comment (tbouffard) [inline tsconfig.json:6]: ❓ during discussions in https://github.com/maxGraph/maxGraph/pull/123#discussion_r990617667 you suggested to switch from es2020 to es6.
Is this something that you want to do in a dedicated PR or did you change your mind?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    "moduleResolution": "node",
+    "strict": true,
+    "esModuleInterop": true,
+    "isolatedModules": true
+  },
+  "paths": {
+    "@maxgraph/core": ["packages/core/src/index.ts"],
+    "@maxgraph/ts-example": ["packages/ts-example/src/main.ts"]
```
Comment (tbouffard) [inline tsconfig.json:14]: ❓ are you sure we need this configuration? according to the documentation, if we keep it, we must also configure `baseUrl`: https://www.typescriptlang.org/tsconfig#paths

File: tsconfig.json (hunk 1 above)
Comment (mayorovad) [reply tsconfig.json:6]: I think it's difficult decision to make. Most modern browsers will work with es2020 just fine, so we can stick with `es2020` for now until supported browsers range is defined.

File: tsconfig.json (hunk 2 above)
Comment (mayorovad) [reply tsconfig.json:14]: Yes, I think you right! Added it to base config and removed from `core` package in d53ea1a

File: tsconfig.json (hunk 1 above)
Comment (tbouffard) [reply tsconfig.json:6]: OK, that's make sense.

**Review by tbouffard — APPROVED** [review]
Nice, thanks for contributing!
